cibuildwheel/linux.py

```diff
@@ -47,6 +47,241 @@ def get_python_configurations(
     ]
 
 
+def get_linux_platforms(
+    options: BuildOptions, python_configurations: List[PythonConfiguration]
+) -> Iterator[Tuple[List[PythonConfiguration], str, str]]:
```

Comment:
I'm pondering naming… so this function is returning a list of Docker containers and the pythons to build within them. So would `get_build_containers` (or `get_configs_by_container`) be a good name? And return a NamedTuple `BuildContainer` with attributes `docker_image`, `python_configurations`, `platform_tag`?

Reply:
+1 on the NamedTuple, return values are a great place to use them. Some of our NamedTuple's really should be dataclasses, but return values are a place where NamedTuple is correct.

I didn't think much about naming. I thought about it just enough to get GitHub Copilot to guess what I meant and write the functions for me because I'm lazy. ;)

Reply:
(Since I'm mostly moving things, it wasn't that helpful; in Hist, it's been pretty impressive to watch it learn about what I'm doing, and then start writing `from_*` methods, special methods, and tests!)

Reply:
I've been in for a while, but haven't started playing with it until last week or so. Had to get into VSCode (which I'm actually really liking, now that I've got the VI extension & with the terminal enabled - and the extension even works in GitHub's "." key editor, which is great).